### DISCUSSION TITLE
How to get Battery Status in transmission via LoRa

### DISCUSSION THREAD

**Discussion by robeweber** [cyberman54/ESP32-Paxcounter#836] [Q&A] [answered]
I am attempting to send the battery status along via lora.

I assume it is an easy one.

In the readme I read the following:

Output of sensor and peripheral data is internally switched by a bitmask register. Default mask can be tailored by editing cfg.payloadmask initialization value in configmanager.cpp following this scheme:

Bit Sensordata Default
0 Paxcounter on
1 unused off
2 BME280/680 on
3 GPS* on
4 User sensor 1 on
5 User sensor 2 on
6 User sensor 3 on
7 Batterylevel off

(I have changed GPSPORT 1 in paxcounter.conf to enable).

_I assume that I have to change something in the lines 9 to 13 in configmanager.cpp:
// default settings for device data to be sent
#define PAYLOADMASK                                                            \
  ((GPS_DATA | MEMS_DATA | COUNT_DATA | SENSOR1_DATA | SENSOR2_DATA |          \
    SENSOR3_DATA) &                                                            \
   (~BATT_DATA) & (~RESERVED_DATA))_

I just can't figure out how to achieve this. Can anybode point me in the right direction?

**Comment by cyberman54** [comment]
You don't need to change the code for this.
You can set transmission of battery status while device is running by sending (downlink) rcommand code 0x14 0x80 0x21.

0x21 is optional. It makes this new setting permanent, by saving in NVRAM. If you do not want this, omit it.

If you need to change the factory defaults in code, this would be
```
#define PAYLOADMASK
((GPS_DATA | MEMS_DATA | COUNT_DATA | SENSOR1_DATA | SENSOR2_DATA |
SENSOR3_DATA | BATT_DATA) & & (~RESERVED_DATA))
```

Note: before reflashing the new build, you need to clear NVRAM. Otherwise the device will pick the previous settings from NVRAM during reboot.

`pio run -t erase`

**Reply by robeweber** [reply]
Thanks cyberman54.

I have tried and received an error message.
Is it not this instead? (this worked without error):
Note the single & instead of double & &.

#define PAYLOADMASK
((GPS_DATA | MEMS_DATA | COUNT_DATA | SENSOR1_DATA | SENSOR2_DATA |
SENSOR3_DATA | BATT_DATA) & (~RESERVED_DATA))

**Comment by cyberman54** [comment, marked as answer]
yes, single &
